miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -8,10 +8,14 @@ import com.rakuten.tech.mobile.miniapp.storage.MiniAppStorage
 
 internal class MiniAppDownloader(
     private val storage: MiniAppStorage,
-    private val apiClient: ApiClient,
+    private var apiClient: ApiClient,
     private val miniAppStatus: MiniAppStatus
 ) {
 
+    fun setApiClient(apiClient: ApiClient) {
```

Comment:
Let's place the update methods at the bottom of the class?

Reply:
I intended to create a interface for any component required to update `ApiClient`

```
/**
 * The interface for MiniApp components which require api interaction.
*/
internal interface MiniAppApiImpl {
    /** Is executed when there is a change of SDK configuration at runtime **/
    fun updateApiClient(apiClient: ApiClient)
}
```

What do you think?

Reply:
It is a good approach too. 👍 

In the current state, because RealMiniApp already is aware of `MiniAppDownloader` & `MiniAppInfoFetcher`, it might not bring any added advantage (i.e., if you want to do something like- for all implementers -> update ApiClient, as this will make you maintain a collection of all the implementers).

But, on the other hand, it can help the new classes which intend to use ApiClient. As they will be aware that the ApiClient can get updated by virtue of this contract.